shared_model/validators/transactions_collection/unsigned_transactions_collection_validator.hpp

```diff
@@ -15,9 +15,15 @@ namespace shared_model {
      * Unsigned transactions collection validator allows to some transaction
      * from the collection to be unsigned. Batch logic should be checked
      */
+    template <typename TransactionValidator>
     class UnsignedTransactionsCollectionValidator
-        : public TransactionsCollectionValidator {
+        : public TransactionsCollectionValidator<TransactionValidator> {
+     private:
+      using ParentType = TransactionsCollectionValidator<TransactionValidator>;
```

Comment:
This is used twice in the code, which can lead to mistakes in case we want to change names later... Maybe, it's better to define this type as ```protected``` in the base class? It also accepts template parameter etc, and there will be no code duplication as well